database/manager/manager.go

```diff
@@ -0,0 +1,238 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package manager
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/leveldb"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/database/meterdb"
+	"github.com/ava-labs/avalanchego/database/prefixdb"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/version"
+)
+
+var (
+	errNonSortedAndUniqueDBs = errors.New("managed databases were not sorted and unique")
+)
+
+type Manager interface {
+	// Current returns the database with the current database version
+	Current() *SemanticDatabase
+	// Last returns the last database prior to the current database or an error if none exists
+	Last() (*SemanticDatabase, bool)
+	// GetDatabases returns all the managed databases in order from current to the oldest version
+	GetDatabases() []*SemanticDatabase
+	// Close closes all of the databases
+	Close() error
+
+	NewPrefixDBManager([]byte) Manager
+	NewNestedPrefixDBManager([]byte) Manager
+	NewMeterDBManager(string, prometheus.Registerer) (Manager, error)
+}
+
+type manager struct {
+	databases []*SemanticDatabase
+}
+
+func (m *manager) Current() *SemanticDatabase { return m.databases[0] }
+
+func (m *manager) Last() (*SemanticDatabase, bool) {
+	if len(m.databases) < 2 {
+		return nil, false
+	}
+	return m.databases[1], true
+}
+
+func (m *manager) GetDatabases() []*SemanticDatabase { return m.databases }
+
+func (m *manager) Close() error {
+	errs := wrappers.Errs{}
+
+	for _, db := range m.databases {
+		errs.Add(db.Close())
+	}
+
+	return errs.Err
+}
+
+// wrapManager returns a new database manager with each managed database wrapped by
+// the [wrap] function. If an error is returned by wrap, the error is returned
+// immediately. If [wrap] never returns an error, then wrapManager is guaranteed to
+// never return an error.
+func (m *manager) wrapManager(wrap func(db *SemanticDatabase) (*SemanticDatabase, error)) (*manager, error) {
+	databases := make([]*SemanticDatabase, len(m.databases))
+	for i, db := range m.databases {
+		wrappedDB, err := wrap(db)
+		if err != nil {
+			return nil, err
+		}
+		databases[i] = wrappedDB
+	}
+	return &manager{databases: databases}, nil
+}
+
+// NewDefaultMemDBManager returns a database manager with a single memory db instance
+// with a default version of v1.0.0
+func NewDefaultMemDBManager() Manager {
+	return &manager{
+		databases: []*SemanticDatabase{
+			{
+				Database: memdb.New(),
+				Version:  version.NewDefaultVersion(1, 0, 0),
+			},
+		},
+	}
+}
+
+// New creates a database manager at [filePath] by creating a database instance from each directory
+// and is less than or equal to the specified version
```

Comment:
`creating a database instance from each directory and is less than or equal to the specified version` 
I don't understand this...missing a few words, maybe?